extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -434,6 +440,127 @@ export class MigrationStateModel implements Model, vscode.Disposable {
+		if (result.connected && result.connectionId) {
+			return azdata.connection.getConnectionString(result.connectionId, true);
+		}
+
+		return '';
+	}
+
+	private updateLoginMigrationResults(newResult: mssql.StartLoginMigrationResult): void {
+		if (this._loginMigrationsResult && this._loginMigrationsResult.exceptionMap) {
+			for (var key in newResult.exceptionMap) {
+				this._loginMigrationsResult.exceptionMap[key] = [...this._loginMigrationsResult.exceptionMap[key] || [], newResult.exceptionMap[key]]
+			}
+		} else {
+			this._loginMigrationsResult = newResult;
+		}
+	}
+
+	public async migrateLogins(): Promise<Boolean> {
+		try {
+			const sourceConnectionString = await this.getSourceConnectionString();
+			const targetConnectionString = await this.getTargetConnectionString();
+			console.log('Starting Login Migration at: ', new Date());
+
+			console.time("migrateLogins")
+			var response = (await this.migrationService.migrateLogins(
+				sourceConnectionString,
+				targetConnectionString,
+				this._loginsForMigration.map(row => row.loginName),
+				this._aadDomainName
+			))!;
+			console.timeEnd("migrateLogins")
+
+			this.updateLoginMigrationResults(response)
+		} catch (error) {
+			console.log('Failed Login Migration at: ', new Date());
+			logError(TelemetryViews.LoginMigrationWizard, 'StartLoginMigrationFailed', error);
+		}
+
+		// TODO AKMA : emit telemetry
+		return true;
+	}
+
+	public async establishUserMappings(): Promise<Boolean> {
+		try {
+			const sourceConnectionString = await this.getSourceConnectionString();
+			const targetConnectionString = await this.getTargetConnectionString();
+
+			console.time("establishUserMapping")
+			var response = (await this.migrationService.establishUserMapping(
+				sourceConnectionString,
+				targetConnectionString,
+				this._loginsForMigration.map(row => row.loginName),
+				this._aadDomainName
+			))!;
+			console.timeEnd("establishUserMapping")
+
+			this.updateLoginMigrationResults(response)
+		} catch (error) {
+			console.log('Failed Login Migration at: ', new Date());
+			logError(TelemetryViews.LoginMigrationWizard, 'StartLoginMigrationFailed', error);
+		}
+
+		// TODO AKMA : emit telemetry
+		return true;
+	}
+
+	public async migrateServerRolesAndSetPermissions(): Promise<Boolean> {
+		try {
+			const sourceConnectionString = await this.getSourceConnectionString();
+			const targetConnectionString = await this.getTargetConnectionString();
+
+			console.time("migrateServerRolesAndSetPermissions")
+			var response = (await this.migrationService.migrateServerRolesAndSetPermissions(
+				sourceConnectionString,
+				targetConnectionString,
+				this._loginsForMigration.map(row => row.loginName),
+				this._aadDomainName
+			))!;
+			console.timeEnd("migrateServerRolesAndSetPermissions")
+
+			this.updateLoginMigrationResults(response)
+
+			console.log('Ending Login Migration at: ', new Date());
```

Comment:
are all these console outputs for debugging, can they be removed

Reply:
yup for debugging. will remove them before merging. leaving these comments open for now as a reminder